base/math.jl

```diff
@@ -293,10 +293,10 @@ end
 
 # polynomial evaluation using compensated summation.
 # much more accurate, especially when lo can be combined with other rounding errors
-@inline function exthorner(x, p::Tuple)
+Base.@assume_effects :terminates_locally  @inline function exthorner(x, p::Tuple)
     hi, lo = p[end], zero(x)
     for i in length(p)-1:-1:1
-        pi = p[i]
+        pi = getfield(p, i)
```

Comment:
Add a comment to explain that this `getfield` is necessary to achieve `:consistent`-cy.